File: chrome/test/data/webui/downloads/downloads_browsertest.js
// Copyright 2015 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/** @fileoverview Tests for the Material Design downloads page. */

// Polymer BrowserTest fixture.
GEN_INCLUDE(['//chrome/test/data/webui/polymer_browser_test_base.js']);

GEN('#include "content/public/test/browser_test.h"');

var DownloadsTest = class extends PolymerTest {
  /** @override */
  get browsePreload() {
    return 'chrome://downloads';
  }
};

var DownloadsItemTest = class extends DownloadsTest {
  /** @override */
  get browsePreload() {
    return 'chrome://downloads/test_loader.html?module=downloads/item_tests.js';
  }
};

TEST_F('DownloadsItemTest', 'All', function() {
  mocha.run();
});

var DownloadsManagerTest = class extends DownloadsTest {
  /** @override */
  get browsePreload() {
    return 'chrome://downloads/test_loader.html?module=downloads/manager_tests.js';
  }
};

TEST_F('DownloadsManagerTest', 'All', function() {
  mocha.run();
});

var DownloadsToolbarTest = class extends DownloadsTest {
  /** @override */
  get browsePreload() {
    return 'chrome://downloads/test_loader.html?module=downloads/toolbar_tests.js';
  }
};

TEST_F('DownloadsToolbarTest', 'All', function() {
  mocha.run();
});

var DownloadsUrlTest = class extends DownloadsTest {
  /** @override */
  get browsePreload() {
    return 'chrome://downloads/a/b/';
  }

  /** @override */
  get extraLibraries() {
    return [
      '//third_party/mocha/mocha.js',
      '//chrome/test/data/webui/mocha_adapter.js',
    ];
  }
};

TEST_F('DownloadsUrlTest', 'All', async function() {
  await import('chrome://webui-test/mojo_webui_test_support.js');
  suite('loading a nonexistent URL of /a/b/', function() {
    test('should load main page with no console errors', function() {
      return customElements.whenDefined('downloads-manager').then(() => {
        assertEquals('chrome://downloads/', location.href);
      });
    });
  });
  mocha.run();
});

var DownloadsSearchServiceTest = class extends DownloadsTest {
  /** @override */
  get browsePreload() {
    return 'chrome://downloads/test_loader.html?module=downloads/search_service_test.js';
  }
};

TEST_F('DownloadsSearchServiceTest', 'All', function() {
  mocha.run();
});

File: chrome/test/data/webui/downloads/downloads_browsertest.cc
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/test/scoped_feature_list.h"
#include "chrome/common/webui_url_constants.h"
#include "chrome/test/base/web_ui_mocha_browser_test.h"
#include "content/public/test/browser_test.h"

class DownloadsTest : public WebUIMochaBrowserTest {
 protected:
  DownloadsTest() { set_test_loader_host(chrome::kChromeUIDownloadsHost); }
};

IN_PROC_BROWSER_TEST_F(DownloadsTest, DangerousDownloadInterstitial) {
  RunTest("downloads/bypass_warning_confirmation_interstitial_test.js",
          "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(DownloadsTest, Item) {
  RunTest("downloads/item_test.js", "runMochaSuite('ItemTest')");
}

IN_PROC_BROWSER_TEST_F(DownloadsTest, Manager) {
  RunTest("downloads/manager_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(DownloadsTest, Toolbar) {
  RunTest("downloads/toolbar_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(DownloadsTest, SearchService) {
  RunTest("downloads/search_service_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(DownloadsTest, NonExistentUrl) {
  // Invoking the test from a non existent URL chrome://downloads/a/b/.
  set_test_loader_host(std::string(chrome::kChromeUIDownloadsHost) + "/a/b");
  RunTestWithoutTestLoader("downloads/non_existent_url_test.js", "mocha.run()");
}
